src/fu-engine.c

```diff
@@ -6385,8 +6387,13 @@ fu_engine_load (FuEngine *self, FuEngineLoadFlags flags, GError **error)
 	/* set up backends */
 	for (guint i = 0; i < self->backends->len; i++) {
 		FuBackend *backend = g_ptr_array_index (self->backends, i);
-		if (!fu_backend_setup (backend, error))
-			return FALSE;
+		g_autoptr(GError) error_backend = NULL;
```

Comment:
I think you need to special case 1 backend. If compiled with one backend it should be fatal no?

Reply:
Would it be fair to say that we should return a failure if they _all_ fail? In the case of win32 with just gusb we'd fail as you suggest, and for Linux embedded it's unlikely we'd fail on embedded without bluez.

Reply:
Yeah I think that's reasonable.

Reply:
Okay, something like that^^?